Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift

```diff
@@ -109,15 +121,81 @@ final class FunctionDeclarationVisitorSpec: QuickSpec {
           expect(self.sut.functionDeclarations.count) == 2
         }
         it("finds the first function") {
-          expect(self.sut.functionDeclarations.first?.name) == "minMax"
-          expect(self.sut.functionDeclarations.first?.returnType) == .tuple([.simple(name: "Int"), .simple(name: "Int")])
+          let firstFunction = self.sut.functionDeclarations.first
+          expect(firstFunction?.name) == "minMax"
+          expect(firstFunction?.arguments?.first?.externalLabel) == "array"
+          expect(firstFunction?.arguments?.first?.type) == .array(element: .simple(name: "Int"))
+          expect(firstFunction?.returnType) == .tuple([.simple(name: "Int"), .simple(name: "Int")])
         }
         it("finds the last function") {
-          expect(self.sut.functionDeclarations.last?.name) == "printWithoutCounting"
-          expect(self.sut.functionDeclarations.last?.returnType).to(beNil())
+          let lastFunction = self.sut.functionDeclarations.last
+          expect(lastFunction?.name) == "printWithoutCounting"
+          expect(lastFunction?.arguments?.first?.externalLabel) == "string"
+          expect(lastFunction?.arguments?.first?.type) == .simple(name: "String")
+          expect(lastFunction?.returnType).to(beNil())
+        }
+      }
+
+      context("function with no external parameter label") {
+        beforeEach {
+          let content = """
+            func print(_ string: String) {
+                let _ = printAndCount(string: string)
+            }
+            """
+
+          try? self.sut.walkContent(content)
+        }
+
+        it("finds the parameter label") {
+          expect(self.sut.functionDeclarations.first?.arguments?.first?.externalLabel) == "_"
+        }
+        it("calculates the selectorName") {
+          expect(self.sut.functionDeclarations.first?.selectorName) == "print(_:)"
+        }
+      }
+
+      context("function with custom external parameter label") {
+        beforeEach {
+          let content = """
+            func append(externalLabel label: String) {
+              // ...
+            }
+            """
+
+          try? self.sut.walkContent(content)
+        }
+
+        it("finds the parameter label") {
+          expect(self.sut.functionDeclarations.first?.arguments?.first?.externalLabel) == "externalLabel"
+        }
+        it("calculates the selectorName") {
+          expect(self.sut.functionDeclarations.first?.selectorName) == "append(externalLabel:)"
+        }
+      }
+
+      context("function with multiple parameters") {
```

Comment:
I think we're missing one test with no parameters

Reply:
09ff3f5